Understanding FF UI

pinkeshmars opened the pull request:
### Description
Understanding FF UI

Linear ticket and [magic word](https://linear.app/docs/github#link-prs) Fixes DEVR-314  

## Type of change
- [ ] Typo fix 
- [X] New feature 
- [ ] Removed outdated references

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,234 @@
+
+From here, you can view the code for your project, connect to GitHub, and also download the source code for your project.
+
+## 11.1 View Code
```
11.1 should be one heading level below 11, here both 11 heading is same level as 11.1

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,234 @@
+
+This option lets you display the *Dart* code for all the pages of your FlutterFlow project. You can also take a look at the dependencies being used by the app here.
+
+## 11.2 Connect GitHub Repo (Paid Feature)
```
Same for 11.2

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,234 @@
+You can use the Preview mode to quickly try out your app on a virtual device without waiting for it to build. This helps you to have a good overview of the navigation and animations you have added to your app. You can also preview your app in the Dark/Light mode and visualize it on various mobile, tablet, and desktop devices.
+
+This mode has a few limitations; you can't use Firebase or perform API calls in the *Preview* mode. However, the onDoubleTap, onLongPress, and onSubmit works in this mode.
+
```
Link to the Preview section of How to Test app page from Projects Section

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,234 @@
+You can also download the code using *[FlutterFlow CLI](https://pub.dev/packages/flutterflow_cli)*. See instructions [here](#).
+
+## 12. Share project
+
```
Somewhere here add a link to the Collaboration page from Projects Section

pinkeshmars replied:
Hey @PoojaB26 I think both are different topics. In collaboration, you add users while this section talks about making your project public.

PoojaB26 replied:
Oh okay, sorry, nvm!

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,234 @@
+
+You can download the entire codebase of the app generated by FlutterFlow using this option. It will download a `.zip` file. You can open the contents using any code editor to make modifications.
+
+### 11.4 Download APK (Paid Feature)
```
Add the "Paid Feature" as a note (Admonitions) below.

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,234 @@
+
+From there, you can configure your project to add GitHub integration to it. You can get a detailed [step-by-step guide](#) for connecting to GitHub.
+
+### 11.3 Download Code (Paid Feature)
```
Add the "Paid Feature" as a note (Admonitions) below.

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,234 @@
+
+![Optimization.avif](imgs/Optimization.avif)
+
+## 9. Project issues
```
Can you please mention the difference between errors and warnings, conceptually and also in terms of compilation. Can be a short text.

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,234 @@
+
+We scan and suggest enhancements to elevate your app's design and speed. Imagine having a personal consultant for design and performance dedicated to improving the user experience of your app.
+
+Optimizations: We spotlight elements that may slow down your app. For example, a query on a column, unused and duplicate backend query, a widget with unbounded size, etc.
```
Make bold - "**Optimizations:**"

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,234 @@
+
+Optimizations: We spotlight elements that may slow down your app. For example, a query on a column, unused and duplicate backend query, a widget with unbounded size, etc.
+
+UI Enhancements: We offer tips for a more visually appealing and user-friendly design, like resizing a small tap target of any widget.
```
Same. Add bold styling to the title

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,234 @@
+
+Restoring the previous version will preserve the current version, then load the changes from the version you're restoring. Before restoration, you may want to view the changes in the previous version. To do this, select the Peek option, which opens the previous version in a new tab.
+
+![Screenshot 2024-05-06 at 10.57.51 AM.png](https://prod-files-secure.s3.us-west-2.amazonaws.com/09694bda-16f5-490f-aaac-3601b64ca6a2/62cabaee-ff75-496b-be7e-6df40a3cd8f2/Screenshot_2024-05-06_at_10.57.51_AM.png)
```
Image doesnt load

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,234 @@
+</div>
+### 6.2 Restoring a version
+
+Restoring the previous version will preserve the current version, then load the changes from the version you're restoring. Before restoration, you may want to view the changes in the previous version. To do this, select the Peek option, which opens the previous version in a new tab.
```
When mentioning FlutterFlow terms like "Peek option", you can add a bold styling, to **Peek** option so it stands out. Specially when you are mentioning it for the first time.

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,234 @@
+
+:::
+
+### 6.1 Saving a version
```
Add the 6.1 and 6.2 steps after you mention Versions at line 100 and describe it. Snapshots can be mentioned after the Version part is over.

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,234 @@
+</div>
+## 6. Project history
+
+Versions
```
Add heading

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,234 @@
+
+Versions enable you to save specific named states of your project. This function is particularly useful for recovery purposes. If you accidentally delete a page or component or change settings, you can easily revert to a saved version.
+
+Snapshots
```
Add heading

PoojaB26 commented on docs/intro/ff-ui/storyboard.md:
```diff
@@ -0,0 +1,58 @@
+# Storyboard
+
+The Storyboard view allows you to visualize the overall design and navigation of your app. On Storyboard, you can see different screens and user interactions that make up your app. This allows you to see how users will navigate through your app and ensure that the user experience is as intuitive and user-friendly as possible.
+
```
A little sentence cleanup from ChatGPT: 

With so many pages displayed on a storyboard, it may be difficult to identify which routes are coming and where they go from a specific page, especially when lines overlap each other. To highlight the incoming and outgoing routes on a specific page, simply click on the page, and the routes will be highlighted in blue.

PoojaB26 commented on docs/intro/ff-ui/canvas.md:
```diff
@@ -0,0 +1,135 @@
+    width: '100%'
+}}>
+    <iframe 
+        src="https://demo.arcade.software/ViB374dOSglo4Z1afa0o?embed&show_copy_link=true"
```
This is a lot of zoom, specially in terms of zoom size. It's pixelating as you zoom. I think a small video clip in between the slides as you change the language will give an idea to the users that text changed. 

And If we want to zoom here, we can do a minimal zoom size change and stay there throughout without moving too much.

PoojaB26 commented on docs/intro/ff-ui/builder.md:
```diff
@@ -0,0 +1,27 @@
+
+## 4. Properties Panel
+
+The Properties Panel lets you modify both the visual appearance and interactive behavior of UI elements on the canvas. It allows you to add Actions, set up a Backend Query, add Animations and more.
```
Is there no page for Properties Panel?

pinkeshmars replied:
I had added it earlier, but since we are not talking about specific widgets or page properties, I just kept the brief here.

PoojaB26 commented on docs/intro/ff-ui/storyboard.md:
```diff
@@ -0,0 +1,58 @@
+
+# Storyboard
+
+The Storyboard view allows you to visualize the overall design and navigation of your app. On Storyboard, you can see different screens and user interactions that make up your app. This allows you to see how users will navigate through your app and ensure that the user experience is as intuitive and user-friendly as possible.
```
```suggestion
With so many pages displayed on a storyboard, it may be difficult to identify which routes are coming and where they go from a specific page, especially when lines overlap each other. To highlight the incoming and outgoing routes on a specific page, simply click on the page, and the routes will be highlighted in blue.
```

PoojaB26 commented on sidebars.ts:
```diff
@@ -13,7 +13,9 @@ const sidebars: SidebarsConfig = {
-            }],
+            },{type: 'category', label: "FlutterFlow UI", items: [
+                'intro/ff-ui/builder', 'intro/ff-ui/navigation-menu','intro/ff-ui/widget-palette','intro/ff-ui/storyboard','intro/ff-ui/toolbar','intro/ff-ui/canvas','intro/ff-ui/resource-hierarchy'
+            ]}],
```
I think the hierarchy is wrong here. Its 

- Getting Started 
- Quickstart 
- Before Your Begin
- Understanding the FF UI 

But in your case, quickstart, before you begin, ff-ui all are nested within Getting Started. 
Please correct this and also you don't have to list pages manually for the non-intro directories. See how the other directories are set up here.

PoojaB26 reviewed (changes requested):
Did a quick review and left some comments!

pinkeshmars commented:
Hey @johnbhiggins7 and @PoojaB26 I've addressed all the review comments so far! 🙂 (except a couple of them, for which I have added comments individually).

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,249 @@
+
+From here, you can view the code for your project, connect to GitHub, and also download the source code for your project.
+
+### 11.1 View Code
```
We dont have to number these sub headings

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,249 @@
+
+### 11.4 Download APK
+
+Use this to generate a release build of your app. It will automatically download the `.apk` file after the building process is complete.
```
Mention this is just for Android

pinkeshmars replied:
Done!

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,249 @@
+
+:::
+
+## 13. Preview
```
I'm wondering if this is becoming repetitive because John explains this and testing  in How to Run & test projects. Can we simply link it there and reduce the following headers to 1 header and just link to the other page?

pinkeshmars replied:
Ok, updated!

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,249 @@
+
+This option lets you display the *Dart* code for all the pages of your FlutterFlow project. You can also take a look at the dependencies being used by the app here.
+
+### 11.2 Connect GitHub Repo (Paid Feature)
```
Same comment about "Paid feature" being part of Admonition instead. Also add hyperlinks to "paid plan" and link to the pricing page. Hyperlink text should be bold as per the Docs style guide

pinkeshmars replied:
Added this into a single admonition.

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,249 @@
+
+## 4. Command Palette
+
+The Command Palette makes it easy to find and use things in the App Builder. Open the Command Palette by clicking the search button or pressing Cmd/Ctrl + K. Search for anything, and to find out where it's used, click the right arrow – we'll show you every place. To directly access the item, just click on it.
```
Shortcuts mentioned here could be bold

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,249 @@
+
+### Snapshots
+
+Snapshots are automatically saved as you build.
```
We can first describe what snapshots are

pinkeshmars replied:
Done!

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,249 @@
+
+Versions enable you to save specific named states of your project. This function is particularly useful for recovery purposes. If you accidentally delete a page or component or change settings, you can easily revert to a saved version.
+
+#### 6.1 Saving a version
```
dont need numbering in title

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,249 @@
+    </iframe>
+</div>
+
+#### 6.2 Restoring a version
```
dont need numbering in title

pinkeshmars replied:
Done!

PoojaB26 commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,249 @@
+
+## 11. Developer menu
+
+From here, you can view the code for your project, connect to GitHub, and also download the source code for your project.
```
maybe a small one line description of what developer menu is like, this allows developers to access developer tools such as ... like in most cases, start with what it is, and then say what you (as devs) can do with it.

pinkeshmars replied:
Done!

PoojaB26 commented on docs/intro/ff-ui/storyboard.md:
```diff
@@ -0,0 +1,59 @@
+
+:::info
+
+Our *Beta* release of Storyboarding is optimized for projects with 30 or fewer pages.
```
Mention that StoryBoard is currently in beta first and what is the limitation after that, and don't use "our", use the, or FlutterFlow's ...

pinkeshmars replied:
Done!

PoojaB26 commented on docs/intro/ff-ui/storyboard.md:
```diff
@@ -0,0 +1,59 @@
+
+## Highlight routes on a page
+
+With so many pages displayed on a storyboard, it may be difficult to identify which routes are coming and where they go from a specific page, especially when lines overlap each other. To highlight the incoming and outgoing routes on a specific page, simply click on the page, and the routes will be highlighted in blue.
```
Please improve the phrasing here "coming and going" may not sound right

PoojaB26 commented on docs/intro/ff-ui/resource-hierarchy.md:
```diff
@@ -0,0 +1,34 @@
+description: Explore the Resource Hierarchy Overview to understand the correlation between traditional Flutter app components and their equivalents in FlutterFlow.
+---
+
+# Resource hierarchy overview
```
I'm a little unsure about this page and it feels like it is directed at Flutter developers more, but anyone else coming to FlutterFlow will be confused about it. I understand this was there in Leigha's talk but that was talk was also for Flutter devs so I'm not sure if this fits completely here. We can pick and choose and just talk about the components of a FlutterFlow project/app and not have the comparison with Flutter maybe? 

@leigha your thoughts here too please, thanks!

pinkeshmars replied:
So the purpose of this page is to just give a brief overview of how Flutter components are translated into FF and then direct users to dedicated pages in our doc to understand more. I agree that it is more targeted towards the Flutter developers, so maybe we can change its location. @leighajarett wdyt?

PoojaB26 replied:
Maybe we do the same diagram but without the comparison in this particular page and I'm assuming resource hierarchy would give an understanding of all the resource categories we have under Resources folder. So like Projects, Pages, Widgets etc. So this should be unrelated to Flutter. But when we talk about generated code, we can add a comparison there. (it will come later) And it will mostly be a FlutterFlow to Flutter code comparison rather than Flutter to FlutterFlow. Would that make sense?

pinkeshmars replied:
I think if we remove the comparison, keeping the diagram doesn't make sense. Instead, I think we could add a 'Resources' tile on the homepage that redirects users to the resources folder. Also, this page is crucial for anyone transitioning from Flutter, we could move it to a more appropriate location.

PoojaB26 replied:
Sorry I meant that we can have a diagram only for the resources hierarchy of FlutterFlow project, so just removing the Flutter terms from the diagram and keeping the FlutterFlow terms. 

My suggestion to cater to both Flutter + FlutterFlow is to do the following flow: 

1. Show diagram with just FlutterFlow terms and 2 line explanations of the resource hierarchy -- Projects, Page, etc. 
2. Next heading section "Flutter to FlutterFlow" talks about "if you are coming from Flutter...then.." And this section shows the original diagram and the comparisons. 
Wdyt?

pinkeshmars replied:
Got it. Updated accordingly. Thanks.

PoojaB26 commented on docs/intro/ff-ui/builder.md:
```diff
@@ -0,0 +1,28 @@
+description: Explore the App Builder in FlutterFlow, featuring a comprehensive interface with four main sections. Navigation Menu, Toolbar, Canvas, and Properties Panel.
+---
+
+# App Builder
```
Overall comment: Can we fix the sidebar positions of the pages and drop them in order? Use front matter to modify positioning.

PoojaB26 commented on docs/intro/ff-ui/canvas.md:
```diff
@@ -0,0 +1,139 @@
+description: Dive into the versatile Canvas in FlutterFlow, where you can effortlessly design and preview your app’s interface.
+---
+
+# Canvas 
```
Also wherever possible remove the headers if they are just one liner descriptions and make the sub headings to bullets there instead.

pinkeshmars replied:
1. Annotated diagram added.
2. I would not recommend dividing the main canvas image into parts just to have less scroll. You can avoid scrolling by directly clicking on the first item in the right-side TOC.
3. Yes, we can remove headers for one line desc. However, For the 'canvas' page specifically, we have a mix of images interspersed with H2 headings, so it's best to keep it as is.

PoojaB26 commented:
Hey @pinkeshmars did a longer review and left some comments on the PR, mostly its around the empty points and headers which are just one line mentions and dont have a huge significance. Suggested some ways to shorten the pages and remove white spaces and empty sections.
Also, please look into rephrasing things where what comes first, and then comes how. I've suggested some places but if you can improve others, that would be great! 
Also lets use less numbered headers and numbered diagrams!

Thank you!!

pinkeshmars commented on docs/intro/ff-ui/canvas.md:
```diff
@@ -0,0 +1,139 @@
+
+## 5. Add AppBar
+
+From here, you can add an [AppBar](#) to your page. Clicking this button opens a popup displaying different [AppBar styles](#) for you to choose from. Just select any AppBar style from the list and it will appear in the Preview Screen.
```
To redirect users to a detailed guide on customizing app bar syles like [this](https://docs.flutterflow.io/settings-and-integrations/general-settings/navbar-and-appbar/appbar#styling-appbar).

pinkeshmars commented on docs/intro/ff-ui/canvas.md:
```diff
@@ -0,0 +1,139 @@
+    width: '100%'
+}}>
+    <iframe 
+        src="https://demo.arcade.software/1IOtwXpNus8W4dLgdHsm?embed&show_copy_link=true"
```
No, it's not on loop, but I have improved and reduced steps in this arcade now. You can check and let me know.

pinkeshmars commented on docs/intro/ff-ui/canvas.md:
```diff
@@ -0,0 +1,139 @@
+
+From here, you can [create a page using AI](#).
+
+## 5. Add AppBar
```
Done.

pinkeshmars commented on docs/intro/ff-ui/canvas.md:
```diff
@@ -0,0 +1,139 @@
+
+If you've enabled the multi-language feature for your project, you can use this to preview your app in different languages. Just select the language from the dropdown menu, and the text in your app will be translated accordingly.
+
+:::info
```
Done.

pinkeshmars commented on docs/intro/ff-ui/canvas.md:
```diff
@@ -0,0 +1,139 @@
+
+## 9. Display Resize Handle Bars
+
+Enabling the Resize Handle Bars will show the handle bards at the right and bottom sides of the preview screen. This allows you to change the size of the preview screen to better test your app on a screen with a custom resolution and the overall responsiveness of your app.
```
Done

PoojaB26 replied:
Maybe missed the "handle bards" typo.

pinkeshmars commented on docs/intro/ff-ui/canvas.md:
```diff
@@ -0,0 +1,139 @@
+
+## 11. Canvas Settings
+
+### 11.1 Safe Area
```
Ok, numbering removed!

pinkeshmars commented on docs/intro/ff-ui/canvas.md:
```diff
@@ -0,0 +1,139 @@
+
+## Video guide
+
+If you prefer watching a video tutorial, here's the one for you:
```
Done!

pinkeshmars commented on docs/intro/ff-ui/canvas.md:
```diff
@@ -0,0 +1,139 @@
+
+### 11.1 Safe Area
+
+When the safe area is enabled, it ensures that content isn't hidden behind physical features like camera notches, sensors, or rounded edges of the display. You can enable or disable it using the Safe Area switch.
```
Done!

pinkeshmars commented on docs/intro/ff-ui/canvas.md:
```diff
@@ -0,0 +1,139 @@
+
+### 11.2 Resize Snapping
+
+When resizing a widget, you can snap its size to the specific pixel values or a percentage of the screen size. By default, it is set to 1 px. To change this, click here, ensure Resize Snapping is enabled, and enter the pixel or percentage value.
```
Done!

pinkeshmars commented on docs/intro/ff-ui/canvas.md:
```diff
@@ -0,0 +1,139 @@
+
+This enables you to simulate how your app appears when users adjust text scale or accessibility settings on their devices. It's vital for developing an app that's visually appealing and also accessible, especially for those requiring larger text sizes.
+
+## 11. Canvas Settings
```
screenshot added!

pinkeshmars commented on docs/intro/ff-ui/canvas.md:
```diff
@@ -0,0 +1,139 @@
+If you've enabled the multi-language feature for your project, you can use this to preview your app in different languages. Just select the language from the dropdown menu, and the text in your app will be translated accordingly.
+
+:::info
+This becomes really helpful in testing your app design with translated text without having you run your app.
```
Done!

pinkeshmars commented on docs/intro/ff-ui/navigation-menu.md:
```diff
@@ -0,0 +1,80 @@
+
+The left side menu on your project page is called the Navigation Menu It gives you access to the project's user interface builder, widget tree, and storyboard. It helps you navigate pages for adding project features, including the Firestore database, setting up API calls, uploading image assets, adding integrations, and more.
+
+![NavigationMenu.avif](imgs/NavigationMenu.avif)
```
moved this page's content inside the builder page.

pinkeshmars commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,249 @@
+
+Hovering over this section reveals the essential information about your project. Here, you can easily check the project name, the current version of FlutterFlow that you're using, along with its release date. Additionally, it shows the Flutter version that your project is currently running on.
+
+## 2. Help
```
Done!

pinkeshmars commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,249 @@
+
+Errors prevent your app from compiling and running. These must be resolved in order to run the app. They can be due to missing actions, errors in custom code, incorrect data types, and so on.
+
+Warnings, while not preventing compilation, indicate potential issues such as incorrect rules setup or performance problems. Although it's possible to ignore warnings, addressing them can enhance the quality of your app and prevent future issues.
```
Done!

pinkeshmars commented on docs/intro/ff-ui/toolbar.md:
```diff
@@ -0,0 +1,249 @@
+
+## 10. Branching
+
+[Branching](#) allows you to create a separate copy of your existing project to build new things without breaking existing features.
```
Done!

pinkeshmars commented:
Hey @PoojaB26, this is ready for review! Most of the review comments are addressed except a few for which I have left comments🙂.

PoojaB26 commented on docs/intro/ff-ui/canvas.md:
```diff
@@ -0,0 +1,144 @@
+
+## 8. Display Device
+
+You can use this to show the device frame in the preview. This might come in handy if you want to visualize better or capture the user interface on a particular device.
```
Add a small note that this is useful when checking your screen with safe area or how it will look with iPhone or Android notches etc (you can frame it better)

pinkeshmars replied:
Done!

PoojaB26 commented on docs/intro/ff-ui/canvas.md:
```diff
@@ -0,0 +1,144 @@
+
+It allows zoom level adjustments and previews in light or dark mode. It also includes features for multi-language preview, adding App Bar and Nav Bar, simulating larger font visualization, and more.
+
+![canvas area](imgs/canvas-new.avif)
```
Thank you for redoing the graphic again! Love it!

Also lets remove the numbering from the headers since the image doesnt have it anymore.

pinkeshmars replied:
Done!

PoojaB26 commented on docs/intro/ff-ui/builder.md:
```diff
@@ -0,0 +1,47 @@
+
+## 3. Canvas Area {#canvas-area}
+
+In the [Canvas Area](canvas), you can see a preview of your mobile device's screen and build your app page.
```
It isnt just mobile, please fix this!

pinkeshmars replied:
Updated!

PoojaB26 commented on docs/intro/ff-ui/builder.md:
```diff
@@ -0,0 +1,47 @@
+14. **Theme settings**: Customize visual appearance.
+15. **Settings and Integrations**: Access app-related settings and integrations.
+
+## 2. Tool Bar {#toolbar}
```
ToolBar won't have a space in between i think, please fix wherever applicable.

pinkeshmars replied:
Updated!

PoojaB26 commented on docs/intro/ff-ui/canvas.md:
```diff
@@ -0,0 +1,144 @@
+
+![handle-bars](imgs/handle-bars.gif)
+
+## 10. Show or hide Keyboard
```
```suggestion
## 10. Show or Hide Keyboard
```

pinkeshmars replied:
Done!

PoojaB26 commented on docs/intro/ff-ui/canvas.md:
```diff
@@ -0,0 +1,144 @@
+
+## 9. Handle Bars
+
+Enabling the Resize Handle Bars will show the handle bards at the right and bottom sides of the preview screen. This allows you to change the size of the preview screen to better test your app on a screen with a custom resolution and the overall responsiveness of your app.
```
```suggestion
Enabling the Resize Handle Bars will show the handle bars at the right and bottom sides of the preview screen. This allows you to change the size of the preview screen to better test your app on a screen with a custom resolution and the overall responsiveness of your app.
```

pinkeshmars replied:
Done! Thanks.

PoojaB26 commented on docs/intro/ff-ui/widget-palette.md:
```diff
@@ -0,0 +1,32 @@
+
+## 4. Theme Widgets
+
+Theme Widgets allow you to customize the visual appearance of a single widget and then reuse it throughout the app. Once you [create a theme widget](#), you can access it from here.
```
Maybe rephrase this and mention a bit that they are part of the design system and helps you build widgets based on your theme etc. With this definition, its not very clear the diff between components and theme widgets.

pinkeshmars replied:
Done!

PoojaB26 commented on docs/intro/ff-ui/widget-palette.md:
```diff
@@ -0,0 +1,32 @@
+
+## 5. Floating Widget Palette
+
+The floating widget palette gives you quick access to widgets directly from the canvas. This feature is useful for swiftly adding widgets without the need to open the widget palette via the navigation menu.
```
```suggestion
The Floating Widget Palette gives you quick access to widgets directly from the canvas. This feature is useful for swiftly adding widgets without the need to open the widget palette via the navigation menu.
```

pinkeshmars replied:
Done!

PoojaB26 commented on docs/intro/ff-ui/widget-palette.md:
```diff
@@ -0,0 +1,32 @@
+
+## 5. Floating Widget Palette
+
+The floating widget palette gives you quick access to widgets directly from the canvas. This feature is useful for swiftly adding widgets without the need to open the widget palette via the navigation menu.
```
A GIF would be useful here if we are not explaining it anywhere else.

pinkeshmars replied:
Added!

PoojaB26 commented on docs/intro/ff-ui/resource-hierarchy.md:
```diff
@@ -0,0 +1,35 @@
+1. **MyApp to Project**: In Flutter, `MyApp` typically represents the root of your application, where you set up routes and other global configurations. In FlutterFlow, the equivalent is the "Project," which encompasses the entire application you are building, including its configurations and settings. Learn more about creating a project [here](../../resources/Projects/how-to-create-find-organize ...
+2. **MyPage to Page**: `MyPage` in Flutter represents a specific screen in the app. Similarly, In FlutterFlow, each "Page" corresponds to a screen, where you build the layout and functionality specific to that page of the project. Learn more about pages in FlutterFlow [here](../../resources/ui-building-blocks/pages/getting-started#creating-a-page).
+3. **Column, Button, Text to Built-in widgets**: In FlutterFlow, widgets are categorized under "Built-in widgets," which users can drag and drop onto their canvas to build the UI. Learn more about widgets [here](../../resources/ui-building-blocks/overview#widgets).
+4. **Custom widget to Component**: `CustomWidget` in Flutter indicates user-created widgets that serve specific functions not covered by built-in widgets. FlutterFlow translates this into "Component" allowing you to create and use custom components within your projects. Learn more about creating a component [here](#).
```
```suggestion
4. **Custom widget to Component**: `CustomWidget` in Flutter indicates user-defined widgets that serve specific functions not covered by built-in widgets. FlutterFlow translates this into "Component" allowing you to create and use custom components within your projects. Learn more about creating a component [here](#).
```

pinkeshmars replied:
Done!

PoojaB26 commented on docs/intro/ff-ui/resource-hierarchy.md:
```diff
@@ -0,0 +1,35 @@
+---
+
+# Resource hierarchy overview
+
```
Love this page! Thanks for making the changes

pinkeshmars replied:
🙏🏻

PoojaB26 reviewed (approved):
Hey @pinkeshmars overall this is great! have made some small suggestions and requested few small changes. You can make those changes and merge this PR, I'm approving it!

pinkeshmars commented on docs/intro/ff-ui/builder.md:
```diff
@@ -0,0 +1,47 @@
+description: Explore the App Builder in FlutterFlow, featuring a comprehensive interface with four main sections. Navigation Menu, Toolbar, Canvas, and Properties Panel.
+sidebar_position: 0
+---
+
```
Done!
